ts/src/connections/tunnelConnection.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Event } from 'vscode-jsonrpc';
+import { ConnectionStatus } from './connectionStatus';
+import { ConnectionStatusChangedEventArgs } from './connectionStatusChangedEventArgs';
+import { RefreshingTunnelAccessTokenEventArgs } from './refreshingTunnelAccessTokenEventArgs';
+import { RetryingTunnelConnectionEventArgs } from './retryingTunnelConnectionEventArgs';
+
+/**
+ * Tunnel connection.
+ */
+export interface TunnelConnection {
```

Comment:
`extends Disposable` ?

Reply:
`Disposable` has `dispose(): void` while here it's `dispose(): Promise<void>`.

Reply:
Is async disposal really needed? `TunnelConnectionBase.dispose()` currently just returns `Promise.resolve()`.

Reply:
Child classes use it. E.g. `TunnelRelayTunnelHost.dipsoe()` awaits termination of all client sessions and deletion of the tunnel endpoint.